File: course-material/labs/classes/coding-exercises/student/counter/java/Counter.java
public class Counter
{
    private int value; // can never go below zero

    public Counter()
    {
        this(0);
    }
    
    public Counter(int initialValue)
    {
        this.value = initialValue;
    }

    public int getValue()
    {
        return value;
    }

    public void setValue(int value)
    {
        this.value = value;
    }

    public void increment()
    {
        ++value;
    }

    public void reset()
    {
        value = 0;
    }
}

File: course-material/labs/classes/coding-exercises/solution/counter/counter.cpp
#include "counter.h"


Counter::Counter()
    : Counter(0) { }

Counter::Counter(unsigned value) : value(value) { }

unsigned Counter::getValue() const
{
    return this->value;
}

void Counter::setValue(unsigned value)
{
    this->value = value;
}

void Counter::increment()
{
    ++value;
}

void Counter::reset()
{
    value = 0;
}
